test/mailers/invitation_mailer_test.rb

```diff
@@ -1,7 +1,19 @@
 require 'test_helper'
 
 class InvitationMailerTest < ActionMailer::TestCase
-  # test "the truth" do
-  #   assert true
-  # end
+  def test_send_invitation_email
+    invitation = Invitation.create(email: "test@email.com")
+    invitation.update_attributes(code: "eff7df8ccc0cd0256619")
+    email = InvitationMailer.invitation_email(invitation)
+
+    assert_emails 1 do
+      email.deliver_now
+    end
+
+    assert_equal "Welcome to EDGI web monitoring", email.subject
+    assert_equal [ENV.fetch('MAIL_SENDER', 'web-monitoring-db@envirodatagov.org')], email.from
+    assert_equal ["test@email.com"], email.to
+    assert_equal read_fixture('invitation_mail_body.html').join, email.html_part.body.to_s
+    assert_equal read_fixture('invitation_mail_body.txt').join, email.text_part.body.to_s
```

Comment:
Hmmm, I don’t know that it really makes sense to duplicate the whole e-mail as a test fixture and check for exact equality. This will make fixing typos or tweaking the text a real pain, and we’ll have to figure out something more complex if we ever substitute user names into the message.

I think you should probably drop this part of the test, or maybe just look for critical items, like using a regex to make sure it includes the signup URL (`http://web-monitoring-db.test/users/sign_up?invitation=\w+`)

Reply:
👍